src/azure-cli/azure/cli/command_modules/network/custom.py

```diff
@@ -5809,11 +5813,42 @@ def run_network_configuration_diagnostic(cmd, client, watcher_rg, watcher_name,
 # endregion
 
 
+# region CustomIpPrefix
+def create_custom_ip_prefix(cmd, client, resource_group_name, custom_ip_prefix_name, location=None,
+                            cidr=None, tags=None, zone=None, signed_message=None, authorization_message=None,
+                            custom_ip_prefix_parent=None):
+
+    CustomIpPrefix = cmd.get_models('CustomIpPrefix')
+    prefix = CustomIpPrefix(
+        location=location,
+        cidr=cidr,
+        zones=zone,
+        tags=tags,
+        signed_message=signed_message,
+        authorization_message=authorization_message
+    )
+
+    if custom_ip_prefix_parent:
+        try:
+            prefix.custom_ip_prefix_parent = client.get(resource_group_name, custom_ip_prefix_name)
+        except ResourceNotFoundError:
+            raise ResourceNotFoundError("Custom ip prefix parent {} is'nt exist".format(custom_ip_prefix_name))
+
+    return client.begin_create_or_update(resource_group_name, custom_ip_prefix_name, prefix)
```

Comment:
It is a long running operation. Please support `--no-wait` and wrap this line by `sdk_no_wait` function